core/test_integration.py

```diff
@@ -0,0 +1,72 @@
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+
+from selenium import webdriver
+from selenium.webdriver.firefox.service import Service
+from webdriver_manager.firefox import GeckoDriverManager
+import time 
+    
+class TestUserLogin(StaticLiveServerTestCase):
+    '''
+    #Using ChromeDriver
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.browser = webdriver.Chrome()
+        cls.browser.implicitly_wait(10)
+        #cls.wait = WebDriverWait(cls.browser, 10)
+        cls.wait = WebDriverWait(cls.browser, 50)  # Increase wait time to 20 seconds
+    '''
+    
+
+    @classmethod
+    def setUpClass(cls):
+        # Setup code to run once before all tests, e.g., starting a browser
+        cls.browser = webdriver.Firefox(service=Service(GeckoDriverManager().install()))
```

Comment:
I'm a little worried that setting up the browser at the class level might mean that the browser state is saved between tests, meaning one test could affect other tests (i.e. if one tests that log in works, the user will be logged in, which might cause a second test of logins to see that user still logged in, even if the code to log them in didn't work). But at the moment this isn't a problem and doing this at the class level will be faster than rebuilding the browser for each test, so let's keep this for now.

Reply:
django.test.TestCase does a full database flush at the start of each
new test. This means that you don't need to manually delete objects in
tearDown - the next test setUp will make sure the database is clean.

doctests and unittest.TestCase do no database preparations. At the
start of a test, the database will be in whatever state the previous
test left. This could mean some stray data which could cause testing
conflicts if you are not careful, so a cleanup might be good practice.

Yours,
Russ Magee %-)

https://groups.google.com/g/django-users/c/NNzptyVgzvg?pli=1

Reply:
Yeah, I'm aware of the DB flush but I'm not sure browser settings would be flushed. It's not a big enough problem to worry about at the moment though.